src/WPF/ArcGISRuntime.WPF.Viewer/SourceCodeViewer.xaml.cs

```diff
@@ -9,134 +9,161 @@
-                    {
-                        // Get the source text from the file.
-                        source = File.ReadAllText(filepath);
-
-                        // Replace the tag characters so that the html renders correctly.
-                        source = source.Replace("<", "&lt;");
-                        source = source.Replace(">", "&gt;");
-
-                        // Build the html.
-                        source =
-                            htmlStart +
-                            "<code class=\"xml\">" +
-                            source +
-                            "</code>" +
-                            htmlEnd;
-                    }
-                    else
+
+                    return _fullContent;
+                }
+            }
+
+            public SourceCodeFile(string sourceFilePath)
+            {
+                _path = sourceFilePath;
+            }
+
+            private void LoadContent()
+            {
+                // Filepaths for the css and js files used for syntax highlighting.
+
+                string cssPath = System.IO.Path.Combine(App.ResourcePath, "Resources", "SyntaxHighlighting", "highlight.css");
+                string jsPath = System.IO.Path.Combine(App.ResourcePath, "Resources", "SyntaxHighlighting", "highlight.pack.js");
+
+                // Source code of the file.
+                try
+                {
+                    string baseContent = File.ReadAllText(_path);
+
+                    if (_path.EndsWith(".xaml"))
                     {
-                        // Continue looping over other files if file is not .cs or .xaml.
-                        continue;
+                        baseContent = baseContent.Replace("<", "&lt;").Replace(">", "&gt;");
                     }
-                    // Add the source html to the _sourceFiles dictionary.
-                    _sourceFiles[filepath] = source;
 
-                    // Add the file as a selectable item in the combobox.
-                    FileSelection.Items.Add(filepath);
+                    // Build the html.
+                    _fullContent =
+                        htmlStart.Replace("{csspath}", cssPath).Replace("{jspath}", jsPath) +
+                        "<code class=\"csharp\">" +
+                        baseContent +
+                        "</code>" +
+                        htmlEnd;
                 }
-                catch (Exception e)
+                catch (Exception ex)
                 {
-                    // Any files that failed to be read will have error messages printed to the console for debugging.
-                    Console.WriteLine(e.Message);
+                    System.Diagnostics.Debug.WriteLine(ex);
                 }
             }
+        }
+    }
 
-            // Default to the index of the last .cs file.
-            FileSelection.SelectedIndex = csIndex;
+    // Taken from StackOverflow: https://stackoverflow.com/a/2586255/4630559
```

Comment:
I struggled with the browser dependency issues for a long time. I'm glad you found a way to resolve it properly.